File: C/3_cau_lenh_dieu_kien/bai_15.c
/*
Create on Thu Oct 02 2025 8:10
@author: ntdai

#16 Nhap vao he so phuong trinh bac hai ax^2+bx+c=0 in nghiem ra man hinh 
*/

#include "stdio.h"
#include "math.h"

int main(){
	// khai bao
	float a, b, c, x1, x2, delta;
	
	// nhap he so
	printf("nhap a = ");
	scanf("%f", &a);
	printf("nhap b = ");
	scanf("%f", &b);
	printf("nhap c = ");
	scanf("%f", &c);
	
	// tinh toan va xuat ket qua
	if (a==0){
		printf("he so a phai khac 0");
	} else {
		delta = pow(b, 2) - 4*a*c;
		printf("delta = %f\n", delta);
		
		if (delta>0 ){
			x1 = (-b-sqrt(delta))/(2*a);
			x2 = (-b+sqrt(delta))/(2*a);
			printf("pt co 2 nghiem x1 = %f va x2 = %f", x1, x2);
		} else if (delta==0){
			x1 = -b/(2*a);
			printf("pt co nghiem kep x1=x2=%2f", x1);
		} else {
			printf("pt vo nghiem");
		}
	}
}

File: python/part_2/bai_15.py
'''
Create Sat Oct 04 2025 14:50
@author: ntdai

#15 GIẢI PHƯƠNG TRÌNH BẬC HAI ax^2+bx+c=0
'''
import math
# Nhập dữ liệu
a = float(input("Nhập a: "))
b = float(input("Nhập b: "))
c = float(input("Nhập c: "))

if(a!=0):
    # ** -> mũ
    delta = b**2 - 4*a*c 
    if (delta<0):
        print("phương trình vô nghiệm")
    elif(delta==0):
        x = -b/(2*a)
        print("phương trình có nghiệm kép x1=x2=", x)
    else:
        x1 = (-b-math.sqrt(delta))/(2*a)
        x2 = (-b+math.sqrt(delta))/(2*a)
        print("phương trình có 2 nghiệm x1={0} và x2={1}".format(x1, x2))
else:
    print("không phải phương trình bậc 2")